rpm/src/main/java/org/eclipse/packager/rpm/RpmTagValue.java

```diff
@@ -56,12 +77,13 @@ public Optional<String> asString() {
         }
 
         if (this.value instanceof String[]) {
-            final String[] arr = (String[]) this.value;
-            if (arr.length > 0) {
-                return Optional.of(arr[0]);
-            } else {
-                return Optional.empty();
+            final String[] array = (String[]) this.value;
+
+            if (array.length == 1) {
+                return Optional.of(array[0]);
             }
+
+            throw new IllegalArgumentException("Array contains more than one string value");
```

Comment:
I don't think that's fully correct. In the case of the array having zero elements, that message would be incorrect.

I am not sure the new logic is an improvement. Before, definition was similar to `pop` or `first`, with an optional "empty" if there was none. Now, it would need to be "exactly one" …

Reply:
I can revert it.

But, I didn't understand what the purpose of it really was. What is the use case for having a list, but only getting the first element.